### PR TITLE
Handle empty data

### PR DESCRIPTION
**Pull request by joetaylor-harness**
Add graceful handling to <StackedList description strings when data items are not present.

### CODE REVIEW

**Comment by joetaylor-harness** [comment]
Pipelines list now better-handles empty/missing fields in the mockPipelines data. Optional fields will no longer render if empty/undefined. The meter component is refactored to no longer require a full set of 10 bars, instead only needing the bars corresponding to non-empty states. The component then pads out at the left any empty bars:

image

**Comment by joetaylor-harness** [comment]
Executions list (standalone and repository) now handles empty/missing data fields. E.g. 1st item has now success field, 3rd items has no description:

image

**Comment by joetaylor-harness** [comment]
Pull Requests list now handles missing or empty fields in data. E.g. 1st item has now merged state set, no author, no version, no review required set:

image

**Comment by joetaylor-harness** [comment]
Added simple node icon option to NodeGroup component:
image

File: packages/canary/src/components/branch-chooser.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+function Left({ children }: { children: React.ReactNode }) {
+  return <div className="flex flex-grow gap-6 items-center">{children}</div>
+}
+
+function Right({ children }: { children: React.ReactNode }) {
+  return <div className="flex gap-6 items-center">{children}</div>
+}
+
+function Dropdown({ title, items }: DropdownProps) {
```
Comment (abhinavrastogi-harness) [inline packages/canary/src/components/branch-chooser.tsx:25]: dropdown should take onchange/onselect callback props

File: packages/canary/src/components/meter.tsx [hunk 2]
```diff
@@ -16,30 +16,22 @@ interface MeterRootProps {
...
   return (
     <div className={cn('flex h-[19px] gap-[4px] items-stretch', className)}>
-      {data?.map((col, col_idx) => {
-        return (
-          <div
-            key={col_idx}
-            className={cn('flex w-[5px] h-full rounded-[1px]', stateToBgColor[col.state], stateToGlow[col.state])}
-          />
-        )
+      {bars.map((col, col_idx) => {
```
Comment (abhinavrastogi-harness) [inline packages/canary/src/components/meter.tsx:32]: index is not a good key: https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66

File: packages/canary/src/components/treeview.tsx [hunk 3]
```diff
@@ -204,14 +205,15 @@ const TreeIndicator = forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivEle
 TreeIndicator.displayName = 'TreeIndicator'
 
 /* eslint-disable @typescript-eslint/no-empty-object-type */
+/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
```
Comment (abhinavrastogi-harness) [inline packages/canary/src/components/treeview.tsx:208]: can we just remove the unused vars instead of disabling their lint rules?

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**